enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,177 @@
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow a cluster administrator to add a
+Windows compute node as a day 2 operator with a prescribed configuration to an
+installer provisioned OpenShift 4.6 cluster and enable scheduling of Windows
+workloads. The targeted platforms for 4.6 are vSphere, Azure and AWS. The
+cluster has to be configured with Hybrid OVN networking.
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy customer
+requirement of being able to run Windows workloads on OpenShift clusters.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Provide workflows for adding Windows instances provisioned using the Machine
+  API to OpenShift clusters
+* Perform all the required steps within the VM for it to be added to the cluster
+  as an OpenShift worker node
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Installing the container runtime in the Windows node
+* Upgrades (This will be addressed in a separate enhancement)
+  * Windows operating system upgrades
+  * Node management like k8s compute component upgrades, reboots and node
+    draining
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is the Windows
+Machine Config operator (WMCO) and a Windows Machine Config Bootstrapper (WMCB)
+binary. The operator will collect information from the cluster and transfer it
+to the Windows node. The WMCB will then perform the necessary steps on the node
+for it to join the cluster. [WMCB](https://github.com/openshift/windows-machine-config-bootstrapper)
+has already been developed as part of the Windows Containers
+[dev preview](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/tools/ansible/docs/ocp-4-4-with-windows-server.md).
+
+### Justification
+
+The reason we are using a WMCB binary rather than a container image is
+that all Windows container images are packaged with a Windows kernel and Red Hat
+has a policy to not ship 3rd party kernels for support reasons.
+
+### Design Details
+
+The operator image will be packaged with the following binaries:
+* Kubelet
+* CNI plugins
+* Hybrid-overlay
+* WMCB
+* Kube-proxy
+
+WMCO will be published to OperatorHub as a Red Hat operator. The way the cluster
+admin will enable Windows workloads on their cluster would be to first install
+WMCO. The operator expects the cluster admin to create a predertmined Secret in
+its namespace containing a private key that will be used to interact with the
+Windows instance. The operator will check for this Secret during boot up time
+and create a user data Secret which the cluster admin can reference in the
```

Comment:
I think, where the user data secret will be used can be explained in later sentence. Without this, we should tell what Windows MachineSet is

Reply:
That is how I had it prior to the latest change but I decided to go with showing the flow. Moreover the reverse question can then be asked i.e. what should the user put in `userDataSecret` of the `MachineSet` CR.

>Without this, we should tell what Windows MachineSet is

I am in the next paragraph. I have also stated `as described below`.  But if you insist I can swap the paragraphs.